### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by stephenplusplus**
Some errors remain in the generated clients' doc files:

```
> @google-cloud/logging@1.0.6 lint C:\Users\sawch\dev\nodejs-logging
> repo-tools lint --cmd eslint -- src/ samples/ system-test/ test/

lint: Linting files in: C:\Users\sawch\dev\nodejs-logging
lint: Running: eslint src/ samples/ system-test/ test/

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_google_api_label.js
  38:4  error  Insert `⏎`  prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_google_api_monitored_resource.js
  94:4  error  Insert `⏎`  prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_google_logging_type_http_request.js
  87:4  error  Insert `⏎`  prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_google_protobuf_any.js
  121:4  error  Insert `⏎`  prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_google_protobuf_duration.js
  84:4  error  Insert `⏎`  prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_google_protobuf_timestamp.js
  88:4  error  Insert `⏎`  prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_log_entry.js
  128:5  error  'LogEntry' is assigned a value but never used                no-unused-vars
  154:5  error  'LogEntryOperation' is assigned a value but never used       no-unused-vars
  181:5  error  'LogEntrySourceLocation' is assigned a value but never used  no-unused-vars
  183:3  error  Insert `⏎`                                                   prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_logging_config.js
  117:5   error  'LogSink' is assigned a value but never used            no-unused-vars
  128:1   error  Delete `⏎`                                              prettier/prettier
  142:10  error  Insert `,`                                              prettier/prettier
  143:4   error  Insert `,`                                              prettier/prettier
  171:5   error  'ListSinksRequest' is assigned a value but never used   no-unused-vars
  191:5   error  'ListSinksResponse' is assigned a value but never used  no-unused-vars
  211:5   error  'GetSinkRequest' is assigned a value but never used     no-unused-vars
  250:5   error  'CreateSinkRequest' is assigned a value but never used  no-unused-vars
  291:5   error  'UpdateSinkRequest' is assigned a value but never used  no-unused-vars
  312:5   error  'DeleteSinkRequest' is assigned a value but never used  no-unused-vars
  314:3   error  Insert `⏎`                                              prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_logging_metrics.js
   65:5   error  'LogMetric' is assigned a value but never used               no-unused-vars
   74:1   error  Delete `⏎`                                                   prettier/prettier
   83:10  error  Insert `,`                                                   prettier/prettier
   84:4   error  Insert `,`                                                   prettier/prettier
  109:5   error  'ListLogMetricsRequest' is assigned a value but never used   no-unused-vars
  129:5   error  'ListLogMetricsResponse' is assigned a value but never used  no-unused-vars
  144:5   error  'GetLogMetricRequest' is assigned a value but never used     no-unused-vars
  167:5   error  'CreateLogMetricRequest' is assigned a value but never used  no-unused-vars
  191:5   error  'UpdateLogMetricRequest' is assigned a value but never used  no-unused-vars
  206:5   error  'DeleteLogMetricRequest' is assigned a value but never used  no-unused-vars
  208:3   error  Insert `⏎`                                                   prettier/prettier

C:\Users\sawch\dev\nodejs-logging\src\v2\doc\doc_logging.js
   42:5  error  'DeleteLogRequest' is assigned a value but never used                          no-unused-vars
  111:5  error  'WriteLogEntriesRequest' is assigned a value but never used                    no-unused-vars
  121:5  error  'WriteLogEntriesResponse' is assigned a value but never used                   no-unused-vars
  139:5  error  'WriteLogEntriesPartialErrors' is assigned a value but never used              no-unused-vars
  195:5  error  'ListLogEntriesRequest' is assigned a value but never used                     no-unused-vars
  222:5  error  'ListLogEntriesResponse' is assigned a value but never used                    no-unused-vars
  243:5  error  'ListMonitoredResourceDescriptorsRequest' is assigned a value but never used   no-unused-vars
  263:5  error  'ListMonitoredResourceDescriptorsResponse' is assigned a value but never used  no-unused-vars
  292:5  error  'ListLogsRequest' is assigned a value but never used                           no-unused-vars
  312:5  error  'ListLogsResponse' is assigned a value but never used                          no-unused-vars
  314:3  error  Insert `⏎`                                                                     prettier/prettier
```

And in the client, about `todo` comments. I actually started removing these, but wasn't sure how to handle:
```
C:\Users\sawch\dev\nodejs-logging\src\v2\logging_service_v2_client.js
  26:1  warning  Unexpected 'todo' comment  no-warning-comments

C:\Users\sawch\dev\nodejs-logging\src\v2\metrics_service_v2_client.js
  26:1  warning  Unexpected 'todo' comment  no-warning-comments
```

Samples aren't too happy, either:
```
C:\Users\sawch\dev\nodejs-logging\samples\fluent.js
  18:23  error  "express" is not found            node/no-missing-require
  26:32  error  "fluent-logger" is not found      node/no-missing-require
  53:33  error  'next' is defined but never used  no-unused-vars

C:\Users\sawch\dev\nodejs-logging\samples\logs.js
   21:27  error  "@google-cloud/logging" is not found          node/no-missing-require
   66:26  error  "bunyan" is not found                         node/no-missing-require
   69:33  error  "@google-cloud/logging-bunyan" is not found   node/no-missing-require
  103:34  error  "@google-cloud/logging-winston" is not found  node/no-missing-require
  129:33  error  "@google-cloud/logging-bunyan" is not found   node/no-missing-require
  143:34  error  "@google-cloud/logging-winston" is not found  node/no-missing-require
  158:27  error  "@google-cloud/logging" is not found          node/no-missing-require
  192:27  error  "@google-cloud/logging" is not found          node/no-missing-require
  224:27  error  "@google-cloud/logging" is not found          node/no-missing-require
  265:27  error  "@google-cloud/logging" is not found          node/no-missing-require
  336:21  error  Empty block statement                         no-empty

C:\Users\sawch\dev\nodejs-logging\samples\quickstart.js
  20:25  error  "@google-cloud/logging" is not found  node/no-missing-require

C:\Users\sawch\dev\nodejs-logging\samples\sinks.js
   21:27  error  "@google-cloud/logging" is not found  node/no-missing-require
   71:27  error  "@google-cloud/logging" is not found  node/no-missing-require
  100:27  error  "@google-cloud/logging" is not found  node/no-missing-require
  127:27  error  "@google-cloud/logging" is not found  node/no-missing-require
  154:11  error  'results' is defined but never used   no-unused-vars
  166:27  error  "@google-cloud/logging" is not found  node/no-missing-require
  179:11  error  'results' is defined but never used   no-unused-vars

C:\Users\sawch\dev\nodejs-logging\samples\system-test\sinks.test.js
  18:25  error  "@google-cloud/logging" is not found  node/no-missing-require
  33:19  error  't' is defined but never used         no-unused-vars
  37:25  error  't' is defined but never used         no-unused-vars

C:\Users\sawch\dev\nodejs-logging\samples\test\fluent.test.js
  18:28  error  "proxyquire" is not published  node/no-unpublished-require
  19:25  error  "supertest" is not published   node/no-unpublished-require
  20:22  error  "ava" is not published         node/no-unpublished-require
```

And this seems legitimate. I didn't get to the step in the migration doc about this yet:
```
C:\Users\sawch\dev\nodejs-logging\system-test\logging.js
  30:19  error  "../../../system-test/env.js" is not found  node/no-missing-require
```

Overall:
```
✖ 74 problems (72 errors, 2 warnings)
  16 errors, 0 warnings potentially fixable with the `--fix` option.
```

### CODE REVIEW

**Comment by lukesneeringer** [comment]
> And in the client, about todo comments. I actually started removing these, but wasn't sure how to handle:

I intentionally left these as warnings, not errors. I want to see them in the output, but do not want CI to block them.

**Comment by lukesneeringer** [comment]
In response to:

```C:\Users\sawch\dev\nodejs-logging\samples\test\fluent.test.js
  18:28  error  "proxyquire" is not published  node/no-unpublished-require
  19:25  error  "supertest" is not published   node/no-unpublished-require
  20:22  error  "ava" is not published         node/no-unpublished-require
```

In speech and in storage, the directory was `samples/system-test`, and I wrote an `.eslintrc.yml` to suppress this rule. Does logging have _both_ a `samples/test` and a `samples/system-test`, or just one?

If it has both, then I would like to figure out why. If it has one, let's standardize.

**Comment by lukesneeringer** [comment]
I fixed the prettier command in the migration doc, which should fix your "some errors remain".

**Comment by jmdobry** [comment]
> Does logging have both a samples/test and a samples/system-test, or just one?

Some of our samples have both. Tests in `test/` are unit tests, and should execute very quickly. Tests in `system-test/` are system tests, and generally hit some service.

**Comment by stephenplusplus** [comment]
I'm down to these errors now:

```
C:\Users\sawch\dev\nodejs-logging\src\metadata.js
  115:9  warning  Unexpected 'todo' comment  no-warning-comments

C:\Users\sawch\dev\nodejs-logging\src\v2\config_service_v2_client.js
  26:1  warning  Unexpected 'todo' comment  no-warning-comments

C:\Users\sawch\dev\nodejs-logging\src\v2\logging_service_v2_client.js
  26:1  warning  Unexpected 'todo' comment  no-warning-comments

C:\Users\sawch\dev\nodejs-logging\src\v2\metrics_service_v2_client.js
  26:1  warning  Unexpected 'todo' comment  no-warning-comments

C:\Users\sawch\dev\nodejs-logging\samples\test\fluent.test.js
  18:28  error  "proxyquire" is not published  node/no-unpublished-require
  19:25  error  "supertest" is not published   node/no-unpublished-require
  20:22  error  "ava" is not published         node/no-unpublished-require

C:\Users\sawch\dev\nodejs-logging\system-test\logging.js
   30:19  error  "../../../system-test/env.js" is not found  node/no-missing-require
   33:1   error  'describe' is not defined                   no-undef
   48:3   error  'before' is not defined                     no-undef
   59:3   error  'after' is not defined                      no-undef
  120:3   error  'describe' is not defined                   no-undef
  121:5   error  'it' is not defined                         no-undef
  139:5   error  'it' is not defined                         no-undef
  165:5   error  'it' is not defined                         no-undef
  189:5   error  'describe' is not defined                   no-undef
  193:7   error  'before' is not defined                     no-undef
  202:7   error  'it' is not defined                         no-undef
  214:7   error  'it' is not defined                         no-undef
  223:5   error  'describe' is not defined                   no-undef
  226:7   error  'before' is not defined                     no-undef
  235:7   error  'it' is not defined                         no-undef
  243:7   error  'it' is not defined                         no-undef
  253:7   error  'it' is not defined                         no-undef
  268:3   error  'describe' is not defined                   no-undef
  303:5   error  'it' is not defined                         no-undef
  317:5   error  'it' is not defined                         no-undef
  330:5   error  'describe' is not defined                   no-undef
  331:7   error  'before' is not defined                     no-undef
  335:7   error  'it' is not defined                         no-undef
  349:7   error  'it' is not defined                         no-undef
  363:5   error  'it' is not defined                         no-undef
  367:5   error  'it' is not defined                         no-undef
  404:5   error  'it' is not defined                         no-undef
  432:5   error  'it' is not defined                         no-undef
  454:5   error  'it' is not defined                         no-undef
  489:5   error  'it' is not defined                         no-undef
  524:5   error  'it' is not defined                         no-undef
  557:5   error  'it' is not defined                         no-undef
  573:5   error  'it' is not defined                         no-undef
  577:5   error  'it' is not defined                         no-undef
  581:5   error  'it' is not defined                         no-undef
  585:5   error  'it' is not defined                         no-undef
  589:5   error  'it' is not defined                         no-undef
  593:5   error  'it' is not defined                         no-undef
  597:5   error  'it' is not defined                         no-undef
  601:5   error  'it' is not defined                         no-undef
```

**Comment by stephenplusplus** [comment]
> 18:28  error  "proxyquire" is not published  node/no-unpublished-require

These errors confuse me.

File: samples/fluent.js [hunk 1]
```diff
@@ -42,15 +42,15 @@ var report = function (err, req) {
...
+        responseStatusCode: 500,
+      },
+    },
   };
   structuredLogger.emit('errors', payload);
 };
 
 // Handle errors (the following uses the Express framework)
-app.use(function (err, req, res, next) {
+app.use(function(err, req, res) {
```
Comment (stephenplusplus) [inline samples/fluent.js]: @jmdobry I removed the `next` to be compliant with the linter. Good or bad idea?

File: samples/sinks.js [hunk 2]
```diff
@@ -142,21 +145,23 @@ function updateSink (sinkName, filter) {
...
 
   // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/logging/latest/logging/sink?method=setMetadata
-  sink.setMetadata(metadata)
-    .then((results) => {
-      console.log(`Sink ${sinkName} updated.`);
+  sink
+    .setMetadata(metadata)
+    .then(results => {
+      const metadata = results[0];
+      console.log(`Sink ${sinkName} updated:`, metadata);
```
Comment (stephenplusplus) [inline samples/sinks.js]: @jmdobry I modified this line so I wouldn't have to remove `results` from the callback arguments. I can revert if you think it was better before.

File: samples/fluent.js (hunk 1 above)
Comment (jmdobry) [reply samples/fluent.js]: If `next` isn't used, nix it

File: samples/sinks.js (hunk 2 above)
Comment (jmdobry) [reply samples/sinks.js]: What you did is fine.

**Comment by stephenplusplus** [comment]
Leaving this small mess:

```

lint: Linting files in: C:\Users\sawch\dev\nodejs-logging
lint: Running: eslint src/ samples/ system-test/ test/

C:\Users\sawch\dev\nodejs-logging\src\metadata.js
  115:9  warning  Unexpected 'todo' comment  no-warning-comments

C:\Users\sawch\dev\nodejs-logging\system-test\logging.js
  30:19  error  "../../../system-test/env.js" is not found  node/no-missing-require
```

**Pull request merged**